src/mpp/Dec.hpp

```diff
@@ -722,12 +756,31 @@ bool decode_next_drop_arg(BUF& buf, T... t)
 	return decode_next_drop_arg_impl<PATH>(iseq{}, buf, t...);
 }
 
+template <class PATH, class BUF, class... T>
+bool decode_next_impl(BUF& buf, T... t);
+
 template <class PATH, class BUF, class... T>
 bool decode_next(BUF& buf, T... t)
 {
 	if constexpr (PATH::size() == 0)
 		return true;
-	else {
+	else if constexpr (path_item_type(PATH::last()) == PIT_DYN_RAW) {
+		using POP_PATH = typename PATH::pop_back_t;
+		auto&& wrapped_dst = path_resolve_parent(POP_PATH{}, t...);
+		using wrapped_dst_t = std::remove_reference_t<decltype(wrapped_dst)>;
+		static_assert(is_raw_decoded_v<wrapped_dst_t, BUF>);
+		auto&& dst = unwrap(wrapped_dst);
+		using dst_t = std::remove_reference_t<decltype(dst)>;
+		BUF begin = std::get<sizeof...(T) - 1>(std::tie(t...));
+		BUF end = buf;
+		if constexpr (tnt::is_pairish_of_v<dst_t, BUF, BUF>) {
```

Comment:
Isn't this condition guaranteed by the static assert above?

Reply:
Any object wrapped with `mpp::as_raw` will pass the static assert. But only a pair of iterators is supported by decoder. It is needed for an easier extension in the future.